storage/repository/uri.go

```diff
@@ -10,6 +10,23 @@ import (
 	"fyne.io/fyne/v2"
 )
 
+// EqualURI returns true if the two URIs are equal.
+//
+// Since: 2.6
+func EqualURI(t1, t2 fyne.URI) bool {
```

Comment:
Do we need to expose the same public function in two packages? Could we just move the implementation to the root `storage` package and just have that one?

Reply:
We don't have access to the internal uri type in the storage package to do a fast path for comparison.

Reply:
Could this move to `internal/storage` maybe without import loops (not sure)? But I just feel like having two public APIs that are identical isn't ideal. It's one thing in the case of e.g. fyne.Do, where one is a static helper for accessing the implementation from the loaded driver, but these are both just static functions.

Reply:
It's the same with almost all other APIs in the repository package

Reply:
What if we make it a method on the URI interface, e.g. `uri1.Equal(uri2)`? Then we can have the implementation on the implementing type here in the repository package (which maybe just calls this function, but we make it private), and we can avoid having repeat public API definitions

Reply:
Changing the interface would be a breaking change for anyone implementing their own URI

Reply:
We could theoretically add the method and look for that in an interface assertion but using the method also wouldn't work in the case for bindings where we pass it in as a function parameter (general form `(T, T) bool` where T is any type).

Reply:
I don't see any solution around this, sorry. The repository package likely should have been internal from the start. I think people generally treat it as such.

Reply:
Unfortunate, but OK. I guess we can consider moving the repository package to internal in Fyne 3.0

Reply:
Yup. We might consider not adding a since: line to the repository but I don't know.

Reply:
nah, I think every public API should probably have them for consistency, and it is technically a public API (as is any exported symbol within a non-internal package)

Reply:
Good point